src/db/mod.rs

```diff
@@ -275,7 +275,7 @@ pub mod db_engine {
     }
 
     pub fn open_db(path: PathBuf, config: DbConfig) -> anyhow::Result<Db> {
-        Db::open(path, &config).map_err(Into::into)
+        Db::open(path, &config)
```

Comment:
Oh wow, it seems Rust got smarter. What recent feature is this?

Reply:
```
warning: useless conversion to the same type: `anyhow::Error`
   --> src/db/mod.rs:278:32
    |
278 |           Db::open(path, &config).map_err(Into::into)
    |  ________________________________-^^^^^^^^^^^^^^^^^^^
279 | |     }
    | |____- help: consider removing
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
    = note: `#[warn(clippy::useless_conversion)]` on by default
```